include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -33,6 +38,93 @@ namespace execution
 inline namespace __dpl
 {
 
+namespace __internal {
+
+#if _ONEDPL_PREDEFINED_POLICIES
+struct __global_instance_tag {};
+#endif
+
+class alignas(sycl::queue) __queue_holder
+{
+    static_assert(sizeof(sycl::queue) >= sizeof(void*));
+    static_assert(alignof(sycl::queue) >= alignof(void*));
+
+    std::byte __buf[sizeof(sycl::queue)];
+
+    bool __has_queue() const
+    {
+        bool res = nullptr!=reinterpret_cast<void* const&>(*this);
+        std::atomic_signal_fence(std::memory_order_acq_rel); // mitigate possible reordering due to type punning
+        return res;
+    }
+
+  public:
+    template <typename... _Args>
+    __queue_holder(_Args... __args)
+    {
+        new (this) sycl::queue(__args...);
```

Comment:
May be more proper way to use `&__buf[0]` or `std::addressof` instead of `this` in this place and in other places below?

Reply:
I believe `this`, `__buf` and `&__buf[0]` are all interchangeable as the address for placement new.
To my taste, using `this` rather than `__buf` in constructors and casts better conveys the idea that the holder instance is really interpreted as the queue; but it's subjective.

Reply:
ok